doc/specs/permissions.md

```diff
@@ -0,0 +1,77 @@
+# Permissions
+
+## Table of contents
+
+1. Motivation
+2. Business rules
+3. Use cases
+
+## Motivation
+
+The site requires certain pages and abilities to be limited to a sub set of users. A user can both be granted extra permissions and have certain permissions removed. A user typically gains extra permissions when transcending classes but can also be granted permissions explicitly. 
+
+## Business rules
+
+ - If a user lacks the permission to access a page then they must not be able to view it and taken to an error page.
+ - If a user lacks the permission to perform a specific action they must not be able to perform it. 
+ - Any change to a user's permissions from an administrative user must be logged in the user's log, indicating reason for the change and the permissions which were added or removed. 
+
+## Use cases
+
+### 1.0 Grant a user permission
+
+**Primary Actor:** An administrator.
+
+**Brief:**  A user can be granted additional permissions by an administrator so they could perform certain actions. 
+
+**Trigger:** An administrator wants to grant a user an extra permission. 
+
+**Preconditions:**
+
+ - The user carries the permission for editing permissions. 
+
+**Postconditions:** 
+
+ - The extra permissions are granted.
+ - The user's log is updated.
+
+**Basic Flow:**
+
+ 1. The administrator navigates to a user's profile page.
+ 2. The administrator navigates to the user's permissions page.
+ 3. The administrator drags permissions from the user's enabled permissions to the user's enabled permissions. 
+ 4. The administrator saves the changes to the user's permissions. 
+
+**Functional requirements:**
```

Comment:
Don't need these headings unless they are required? Perhaps have a spec template, or a use case template somewhere and keep these in there as a starting point?